<launch>

  <arg name="camera_type"	default="v4l2"/>
  <arg name="camera_name"	default="$(eval 'V4L2Camera' if camera_type ==
					 'v4l2' else 'IIDCCamera')"/>
  <arg name="max_skew"		default="0"/>
  <arg name="convert_to_rgb"	default="false"/>
  <arg name="rate"		default="100"/>
  <arg name="manager"		default=""/>
  <arg name="manager_threads"	default="4"/>
  <arg name="vis"		default="false"/>

  <arg if="$(eval manager == '')"	name="command"	value="standalone"/>
  <arg unless="$(eval manager == '')"	name="command"	value="load"/>

  <node name="$(arg manager)"
	pkg ="nodelet" type="nodelet" args="manager" output="screen">
    <param name="num_worker_threads"	value="$(arg manager_threads)"/>
  </node>

  <node name="$(arg camera_name)"
	pkg = "nodelet" type="nodelet" output="screen"
	args="$(arg command) aist_area_camera/$(arg camera_type)_camera
	      $(arg manager)">
    <param name="camera_name"		value="$(arg camera_name)"/>
    <param name="max_skew"		value="$(arg max_skew)"/>
    <param name="convert_to_rgb"	value="$(arg convert_to_rgb)"/>
    <param name="rate"			value="$(arg rate)"/>
  </node>

  <group if="$(arg vis)">

    <node name="image_view"
  	  pkg ="nodelet" type="nodelet" output="screen"
  	  args="$(arg command) image_view/image $(arg manager)">
      <remap from="image"	to="$(arg camera_name)/camera0/image"/>
    </node>

    <node name="rqt_reconfigure"
	  pkg="rqt_reconfigure" type="rqt_reconfigure"/>

  </group>

</launch>
